cdc/sink/codec/csv/csv_encoder.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csv
+
+import (
+	"bytes"
+	"context"
+	"errors"
+
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/sink/codec"
+	"github.com/pingcap/tiflow/cdc/sink/codec/common"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+)
+
+// BatchEncoder encodes the events into the byte of a batch into.
+type BatchEncoder struct {
+	valueBuf    *bytes.Buffer
+	callbackBuf []func()
+	batchSize   int
+	csvConfig   *config.CSVConfig
+}
+
+// AppendRowChangedEvent implements the EventBatchEncoder interface
+func (b *BatchEncoder) AppendRowChangedEvent(
+	_ context.Context,
+	_ string,
+	e *model.RowChangedEvent,
+	callback func(),
+) error {
+	if b.csvConfig == nil {
```

Comment:
why we need this? It should never be nil logically.

Reply:
This checking is for defensive purpose because we don't prevent caller passing a nil config in `NewBatchEncoderBuilder`